spec/dd-agent_spec.rb

```diff
@@ -205,6 +205,122 @@ def set_env_var(name, value)
     it_behaves_like 'version set below 4.x'
   end
 
+  context 'allows a string for agent version' do
+    cached(:chef_run) do
+      ChefSpec::SoloRunner.new(
+        :platform => 'ubuntu',
+        :version => '14.10'
+      ) do |node|
+        node.set['datadog'] = {
+          'api_key' => 'somethingnotnil',
+          'agent_version' => '4.4.0-200'
+        }
+      end.converge described_recipe
+    end
+
+    it 'installs agent 4.4.0-200' do
+      expect(chef_run).to install_package('datadog-agent').with(version: '4.4.0-200')
+    end
+  end
+
+  context 'allows a hash for agent version' do
+    context 'when ubuntu' do
+      cached(:chef_run) do
+        ChefSpec::SoloRunner.new(
+          :platform => 'ubuntu',
+          :version => '14.10'
+        ) do |node|
+          node.set['datadog'] = {
+            'api_key' => 'somethingnotnil',
+            'agent_version' => {
+              'debian' => '1:5.9.0-1',
+              'rhel' => '4.4.0-200',
+              'windows' => '4.4.0'
+            }
+          }
+        end.converge described_recipe
+      end
+
+      it 'installs agent 1:5.9.0-1' do
+        expect(chef_run).to install_apt_package('datadog-agent').with(version: '1:5.9.0-1')
+      end
+    end
+
+    context 'when windows' do
+      cached(:chef_run) do
+        ChefSpec::SoloRunner.new(
+          :platform => 'windows',
+          :version => '2012R2',
+          :file_cache_path => '/tmp'
```

Comment:
I'd prefer a "realistic" path on Windows (`C:/chef/cache` for instance, see https://github.com/DataDog/chef-datadog/blob/v2.6.0/spec/dd-agent_spec.rb#L162)